### PR TITLE
[YUNIKORN-1052] Add more cluster-wide metrics to Yunikorn

### PR DESCRIPTION
**Pull request by pbacsko**
### What is this PR for?
Add more cluster-wide metrics to Yunikorn.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1052

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/scheduler/objects/node.go [hunk 1]
```diff
@@ -72,9 +81,9 @@ func NewNode(proto *si.NodeInfo) *Node {
 		allocations:       make(map[string]*Allocation),
 		schedulable:       true,
 		listeners:         make([]NodeListener, 0),
+		ready:             ready,
```
Comment (pbacsko) [inline pkg/scheduler/objects/node.go:84]: On this abstraction level, this field could have a different name like `acceptingContainers` or just `healthy`.

File: pkg/scheduler/objects/node.go [hunk 2]
```diff
@@ -49,6 +50,7 @@ type Node struct {
 	allocations       map[string]*Allocation
 	schedulable       bool
 	unlimited         bool
+	ready             bool
```
Comment (pbacsko) [inline pkg/scheduler/objects/node.go:53]: On this abstraction level, this field could have a different name like `acceptingContainers` or just `healthy`.

File: pkg/scheduler/context.go [hunk 3]
```diff
@@ -631,6 +635,23 @@ func (cc *ClusterContext) updateNode(nodeInfo *si.NodeInfo) {
 
 	switch nodeInfo.Action {
 	case si.NodeInfo_UPDATE:
+		var newReadyStatus bool
+		var err error
+		if newReadyStatus, err = strconv.ParseBool(nodeInfo.Attributes["ready"]); err != nil {
+			log.Logger().Error("Could not parse ready attribute, assuming true", zap.Any("attributes", nodeInfo.Attributes))
+			newReadyStatus = true
```
Comment (wilfred-s) [inline pkg/scheduler/context.go:642]: Are we sure that we get this information from the k8shim?
Looking at what is send by the shim we seem to only get the resources and some fixed rack related info

File: pkg/scheduler/objects/node.go (hunk 2 above)
Comment (wilfred-s) [reply pkg/scheduler/objects/node.go:53]: I think so too, isn't it up to us to pick one?

File: pkg/scheduler/context.go (hunk 3 above)
Comment (pbacsko) [reply pkg/scheduler/context.go:642]: Please check https://github.com/apache/incubator-yunikorn-k8shim/pull/362. This is the shim side change.

File: pkg/scheduler/objects/node.go (hunk 2 above)
Comment (wilfred-s) [reply pkg/scheduler/objects/node.go:53]: Checking the shim side and K8s itself I think we need to stick with _Ready_ to not confuse things.

**Review by wilfred-s — APPROVED** [review]
LGTM
went to the code again, with the shim change it looks good

**Comment by wilfred-s** [comment]
@pbacsko can I commit this I don't think we need anything more at the moment?

**Comment by pbacsko** [comment]
> @pbacsko can I commit this I don't think we need anything more at the moment?

Yeah, it's in good shape. I'll do two minor cleanup-related things:
1) Replace "ready" literal with a constant 
2) Simplify tests a bit, there is some code duplication

I'll post an update soon.

File: pkg/scheduler/context.go [hunk 4]
```diff
@@ -640,10 +661,21 @@ func (cc *ClusterContext) updateNode(nodeInfo *si.NodeInfo) {
 	case si.NodeInfo_DRAIN_NODE:
 		// set the state to not schedulable
 		node.SetSchedulable(false)
+		metrics.GetSchedulerMetrics().IncDrainingNodes()
```
Comment (wilfred-s) [inline pkg/scheduler/context.go:664]: oops, I overlooked an error here I think: if the node is already draining we should not increment again and we do not need to change state:
```
if node.IsSchedulable() {
	node.SetSchedulable(false)
	metrics.GetSchedulerMetrics().IncDrainingNodes()
}
```
this is handled the other way around

**Review by wilfred-s — CHANGES_REQUESTED** [review]
late incoming change

File: pkg/scheduler/context.go (hunk 4 above)
Comment (pbacsko) [reply pkg/scheduler/context.go]: Fixed

**Review by wilfred-s — APPROVED** [review]
LGTM

**Pull request closed without merging**